### ISSUE TITLE
[cloud-tensorflow-mnist]: Step 7 requires AdamOptimizer

### ISSUE THREAD

**Issue by TTransmit** [googlecodelabs/feedback#228]
On Step 7 - Lab: adding layers, I found adding layers alone resulted in the network sitting at around 0.2 accuracy before improving late in the run to 0.68. I was confused so looked in the python file `mnist_2.0_five_layers_sigmoid`. I saw that also changed from `GradientDescentOptmizer` to `AdamOptimizer`. Making this change pushed the accuracy to 0.97. The need to or reason for changing optimizers is not mentioned in the lab text. 

Here is code with the change to AdamOptimizer shown

```
import tensorflow as tf
import tensorflowvisu
from tensorflow.examples.tutorials.mnist import input_data as mnist_data
print("Tensorflow version " + tf.__version__)
tf.set_random_seed(0)

mnist = mnist_data.read_data_sets("data", one_hot=True, reshape=False, validation_size=0)

X = tf.placeholder(tf.float32, [None, 28, 28, 1])
Y_ = tf.placeholder(tf.float32, [None, 10])

L = 200
M = 100
N = 60
O = 30

W1 = tf.Variable(tf.truncated_normal([28*28, L] ,stddev=0.1))
B1 = tf.Variable(tf.zeros([L]))

W2 = tf.Variable(tf.truncated_normal([L, M], stddev=0.1))
B2 = tf.Variable(tf.zeros([M]))

W3 = tf.Variable(tf.truncated_normal([M, N] ,stddev=0.1))
B3 = tf.Variable(tf.zeros([N]))

W4 = tf.Variable(tf.truncated_normal([N, O], stddev=0.1))
B4 = tf.Variable(tf.zeros([O]))

W5 = tf.Variable(tf.truncated_normal([O, 10] ,stddev=0.1))
B5 = tf.Variable(tf.zeros([10]))

XX = tf.reshape(X, [-1, 28*28])

Y1 = tf.nn.sigmoid(tf.matmul(XX, W1) + B1)
Y2 = tf.nn.sigmoid(tf.matmul(Y1, W2) + B2)
Y3 = tf.nn.sigmoid(tf.matmul(Y2, W3) + B3)
Y4 = tf.nn.sigmoid(tf.matmul(Y3, W4) + B4)
Y  = tf.nn.softmax(tf.matmul(Y4, W5) + B5)

cross_entropy = -tf.reduce_mean(Y_ * tf.log(Y)) * 1000.0

correct_prediction = tf.equal(tf.argmax(Y, 1), tf.argmax(Y_, 1))
accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))

train_step = tf.train.GradientDescentOptimizer(0.005).minimize(cross_entropy) # remove when using AdamOptimizer version
# Needs the AdamOptimizer to show improved accuracy over the single layer net
# train_step = tf.train.AdamOptimizer(0.005).minimize(cross_entropy)

allweights = tf.reshape(W1, [-1])
allbiases = tf.reshape(B1, [-1])
I = tensorflowvisu.tf_format_mnist_images(X, Y, Y_)
It = tensorflowvisu.tf_format_mnist_images(X, Y, Y_, 1000, lines=25)
datavis = tensorflowvisu.MnistDataVis()

init = tf.global_variables_initializer()
sess = tf.Session()
sess.run(init)

def training_step(i, update_test_data, update_train_data):

    batch_X, batch_Y = mnist.train.next_batch(100)

    if update_train_data:
        a, c, im, w, b = sess.run([accuracy, cross_entropy, I, allweights, allbiases], feed_dict={X: batch_X, Y_: batch_Y})
        datavis.append_training_curves_data(i, a, c)
        datavis.append_data_histograms(i, w, b)
        datavis.update_image1(im)
        print(str(i) + ": accuracy:" + str(a) + " loss: " + str(c))

    if update_test_data:
        a, c, im = sess.run([accuracy, cross_entropy, It], feed_dict={X: mnist.test.images, Y_: mnist.test.labels})
        datavis.append_test_curves_data(i, a, c)
        datavis.update_image2(im)
        print(str(i) + ": ********* epoch " + str(i*100//mnist.train.images.shape[0]+1) + " ********* test accuracy:" + str(a) + " test loss: " + str(c))

    sess.run(train_step, feed_dict={X: batch_X, Y_: batch_Y})

datavis.animate(training_step, iterations=2000+1, train_data_update_freq=10, test_data_update_freq=50, more_tests_at_start=True)

print("max test accuracy: " + str(datavis.get_max_test_accuracy()))
```

**Comment by TTransmit** [comment]
This is explained on the next page of the tutorial. Sorry.
